index.js

```diff
@@ -12,15 +12,16 @@ var net = require("net"),
     parsers = [], commands,
     connection_id = 0,
     default_port = 6379,
-    default_host = "127.0.0.1";
-
-// can set this to true to enable for all connections
-exports.debug_mode = false;
+    default_host = "127.0.0.1",
+    debugModeEnabled = /\bredis\b/i,
+    debug = function () {
+        if (debugModeEnabled.test(process.env.NODE_DEBUG)) {
```

Comment:
Probably can do the debug test outside the debug function, we don't need to support the user changing `proces.env.NODE_DEBUG` during execution.

E.g.
```
var debug = function noop() {};
if (debugModeEnabled.test(process.env.NODE_DEBUG)) {
  debug = function debug() {
    console.error.apply(...);
  }
}
if (util.debuglog) {
  debug = util.debuglog('redis');
}